thali/install/install.js

```diff
@@ -0,0 +1,240 @@
+function copyDevelopmentThaliCordovaPluginToProject(appRootDirectory, thaliDontCheckIn, depotName, branchName) {
+    var targetDirectory = createUnzipedDirectoryPath(depotName, branchName, thaliDontCheckIn);
+    var sourceDirectory = path.join(appRootDirectory, "../Thali_CordovaPlugin");
+    return new Promise(function(resolve, reject) {
+        fs.remove(targetDirectory, function(err) {
+            if (err) {
+                reject("copyDevelopmentThaliCordovaPluginToProject remove failed with " + err);
+                return;
+            }
+           fs.copy(sourceDirectory, targetDirectory, function (err) {
+               if (err) {
+                   reject("copyDevelopmentThaliCordovaPluginToProject failed with" + err);
+                   return;
+               }
+               resolve(createGitHubZipResponse(depotName, branchName, thaliDontCheckIn, true));
+           }); 
+        });
+    });
+}
+
+function doesMagicDirectoryNamedExist(thaliDontCheckIn) {
+    var magicFileLocation = path.join(thaliDontCheckIn, magicDirectoryNameToDoLocalDevelopment);
+    return fs.existsSync(magicFileLocation);
+}
+
+module.exports = function(callBack) {
+   //get the app root folder from app/www/jxcore/node_modules/thali
+    var appRootDirectory = path.join(__dirname, '../../../../../');
+    var thaliDontCheckIn = path.join(appRootDirectory, "thaliDontCheckIn" );
+    var appScriptsFolder = path.join(appRootDirectory, "plugins/org.thaliproject.p2p/scripts");
+    var jxcoreFolder = path.join(appRootDirectory, 'www/jxcore' );
+
+    if(!(path.basename(jxcoreFolder) == 'jxcore')) {
+        callBack('Could not locate JXCore folder. Exiting the thali plugin installation..', null);
+    }
+    
+    var thaliProjectName = "thaliproject";
+    var thaliDepotName = "Thali_CordovaPlugin";
+    var thaliBranchName = "story_0";
+    var getThaliCordovaPluginZip = 
+        doesMagicDirectoryNamedExist(thaliDontCheckIn) ?
+            copyDevelopmentThaliCordovaPluginToProject(appRootDirectory, thaliDontCheckIn, thaliDepotName, thaliBranchName) :
+            installGitHubZip(thaliProjectName, thaliDepotName, thaliBranchName, thaliDontCheckIn);
+
+    getThaliCordovaPluginZip
+    .then(function(thaliCordovaPluginUnZipResult) {      
+        if (thaliCordovaPluginUnZipResult.directoryUpdated) {
+            var weAddedPluginsFile = path.join(thaliDontCheckIn, "weAddedPlugins");
+            return uninstallPluginsIfNecessary(weAddedPluginsFile, appRootDirectory)
+            .then(function() {
+                return childProcessExecPromise('cordova plugin add ' + thaliCordovaPluginUnZipResult.unzipedDirectory, appRootDirectory);   
+            }).then(function() {
+                return childProcessExecPromise('jx npm install', appScriptsFolder);       
+            }).then(function() {
+                return fs.writeFileAsync(weAddedPluginsFile, "yes");
+            });
+        }
+    }).then(function() {
+        callBack(null, null);
```

Comment:
I still like mine better but I recognize idiomatic usage when I see it so I changed it.